extension/test/unit/lambda.spec.js

```diff
@@ -40,6 +40,16 @@ describe('Lambda handler', () => {
     expect(result.actions).to.equal(undefined)
   })
 
+  it('does not throw unhandled exception when handlePayment data is null', async () => {
+    sinon.stub(paymentHandler, 'handlePayment').returns({ success: true, data: null })
+
+    const result = await handler(event)
+
+    expect(result.responseType).equals('UpdateRequest')
```

Comment:
It seems that in the described case it should rather say `RequestIgnored` as there will be no updates applied?

Reply:
Hi @butenkor, as this will be running in a AWS Lambda the only valid `responseType` that commercetools will allow is either `UpdateRequest` or `FailedValidation`. We don't want failed validation in this case because this will cause all updates in the transaction to rollback.

https://docs.commercetools.com/http-api-projects-api-extensions#response

Also in this case we don't know the request is actually ignored, we only know paymentHandler returned a `{success: true, data: null}`, the change is to just handle the case of `{ data: null }`, which currently causes the code to bomb out

Reply:
Agree